mat64/format.go

```diff
@@ -9,15 +9,66 @@ import (
 	"strconv"
 )
 
-// Format prints a pretty representation of m to the fs io.Writer. The format character c
+// Formatted returns a fmt.Formatter for the matrix m using the given options.
+func Formatted(m Matrix, options ...FormatOption) fmt.Formatter {
+	f := formatter{
+		matrix: m,
+		dot:    '.',
+	}
+	for _, o := range options {
+		o(&f)
+	}
+	return f
+}
+
+type formatter struct {
+	matrix Matrix
+	prefix string
+	margin int
+	dot    byte
```

Comment:
>In particular, some people might like to substitute ASCII alternatives.

Yes, as I mentioned I had a play with this. I'm happy to work on that some more, but we can do that later. I'll file an issue for that if this PR is merged.